sprockets/mixins/amqp/__init__.py

```diff
@@ -11,48 +11,610 @@
 
-except ImportError as error:
-    class PublishingMixin(object):
-        def __init__(self, *args, **kwargs):
-            raise error
+__version__ = '2.0.0'
 
-    class AMQP(object):
-        def __init__(self, *args, **kwargs):
-            raise error
+LOGGER = logging.getLogger(__name__)
 
-version_info = (2, 0, 0)
-__version__ = '.'.join(str(v) for v in version_info)
+DEFAULT_RECONNECT_DELAY = 5
+DEFAULT_CONNECTION_ATTEMPTS = 3
 
 
-def install(application, **kwargs):
-    """Call this to install AMQP for the Tornado application.
+def install(application, io_loop=None, **kwargs):
+    """Call this to install AMQP for the Tornado application. Additional
+    keyword arguments are passed through to the constructor of the AMQP
+    object.
 
+    :param tornado.web.Application application: The tornado application
+    :param tornado.ioloop.IOLoop io_loop: The current IOLoop.
     :rtype: bool
 
     """
     if getattr(application, 'amqp', None) is not None:
         LOGGER.warning('AMQP is already installed')
         return False
 
-    if os.environ.get('AMQP_URL'):
-        kwargs['url'] = os.environ['AMQP_URL']
-    if os.environ.get('AMQP_TIMEOUT'):
-        kwargs['timeout'] = int(os.environ['AMQP_TIMEOUT'])
-    if os.environ.get('AMQP_RECONNECT_DELAY'):
-        kwargs['reconnect_delay'] = int(os.environ['AMQP_RECONNECT_DELAY'])
-    if os.environ.get('AMQP_CONNECTION_ATTEMPTS'):
-        kwargs['connection_attempts'] = int(
-            os.environ['AMQP_CONNECTION_ATTEMPTS'])
-
-    setattr(application, 'amqp', AMQP(**kwargs))
+    kwargs.setdefault('io_loop', io_loop)
+
+    # Support AMQP_* and RABBITMQ_* variables
+    for prefix in {'AMQP', 'RABBITMQ'}:
+
+        key = '{}_URL'.format(prefix)
+        if os.environ.get(key) is not None:
+            LOGGER.debug('Setting URL to %s', os.environ[key])
+            kwargs.setdefault('url', os.environ[key])
+
+        key = '{}_CONFIRMATIONS'.format(prefix)
+        if os.environ.get(key) is not None:
+            value = os.environ[key].lower()
+            LOGGER.debug('Setting enable_confirmations to %s', value == 'true')
+            kwargs.setdefault('enable_confirmations', value == 'true')
+
+        key = '{}_CONNECTION_ATTEMPTS'.format(prefix)
+        if os.environ.get(key) is not None:
+            value = int(os.environ[key])
+            LOGGER.debug('Setting connection_attempts to %s', value)
+            kwargs.setdefault('connection_attempts', value)
+
+        key = '{}_RECONNECT_DELAY'.format(prefix)
+        if os.environ.get(key) is not None:
+            value = float(os.environ[key])
+            LOGGER.debug('Setting reconnect_delay to %s', value)
+            kwargs.setdefault('reconnect_delay', value)
+
+    # Set the default AMQP app_id property
+    kwargs['default_app_id'] = '{}/{}'.format(
+        application.settings.get('service'),
+        application.settings.get('version'))
+    if kwargs['default_app_id'] == 'None/None':
```

Comment:
Well, these are the poss. cases, since you're only using default when outcome is None/None

- some-service/some-version
- some-service/None
- None/some-version
- None/None

I'm thinking None is never useful?